File: python/euler-14.py
def check(element):
    e = element
    sayac = 1

    while e > 1:
        if e % 2 == 0:
            e = e // 2
            sayac += 1
        else:
            e = (3 * e) + 1
            sayac += 1
    
    return sayac

max = 0
number = 0

for i in range(3, 1000000):
    chain = check(i)
    if chain > max:
        max = chain
        number = i

print("Longest chain: " + str(max))
print("Number that produced the chain: "+ str(number))

File: js/euler-14.js
function check(element) {
  let e = element;
  let sayac = 1;

  while (e > 1) {
    if (e % 2 === 0) {
      e = e / 2;
      sayac++;
    }
    else {
      e = (3 * e) + 1;
      sayac++;
    }
  }
  return sayac;
}

let foo = 100;
let sayi = 0;

for (let i = 3; i < 1000000; i++) {
  let element = check(i);
  if (element > foo) {
    foo = element;
    sayi = i;
  }
}

console.log(foo);
console.log(sayi);